packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,217 @@
+
+  return (
+    <svg
+      aria-hidden="true"
+      height={svgSizeRem}
+      viewBox={`0 0 ${svgViewBoxSize} ${svgViewBoxSize}`}
+      width={svgSizeRem}
+    >
+      <rect
+        className={styles.outline}
+        fill="none"
+        height="18"
+        rx="3"
+        stroke="currentColor"
+        strokeWidth="2"
+        width="18"
+        x="1"
+        y="1"
+      />
+      <path className={styles.content} d={checkmarkPath} fill="none" />
+    </svg>
+  );
+};
+
+/**
+ * Checkbox input element, exported for greater flexibility.
+ * You must provide an `id` prop and connect it to a visible label.
+ */
+export const CheckboxInput = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ checked, className, disabled, ...rest }, ref) => {
+    // Make indeterminate checkbox visually match the colors of a
+    // checked state, but announce itself as "mixed" to screen readers
+    const checkedProps =
+      checked === "indeterminate"
+        ? {
+            "aria-checked": "mixed" as const,
+            checked: true,
+          }
+        : {
+            checked,
+          };
+
+    return (
+      <span className={clsx(styles.inputWrapper, disabled && styles.disabled)}>
+        <input
+          className={clsx(className, styles.checkboxInput)}
+          disabled={disabled}
+          ref={ref}
+          style={svgStyle}
+          type="checkbox"
+          {...checkedProps}
+          {...rest}
+        />
+        <CheckboxSvg indeterminate={checked === "indeterminate"} />
+      </span>
+    );
+  },
+);
+
+/**
+ * Label element, exported for greater flexibility. Can be used with any form input.
+ */
+export const Label = ({
+  children,
+  className,
+  htmlFor,
+  size = "medium",
+}: LabelProps) => {
+  return (
+    <label
+      className={clsx(
+        className,
+        styles.label,
+        size === "small" && styles.labelSmall,
+        size === "medium" && styles.labelMedium,
+      )}
+      htmlFor={htmlFor}
+      style={svgStyle}
+    >
+      {children}
+    </label>
+  );
+};
+
+/**
+ * Design system Checkbox.
+ */
+const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  (props, ref) => {
+    // All remaining props are passed to the `input` element
+    const { className, id, label, size, ...rest } = props;
+
+    if (
+      process.env.NODE_ENV !== "production" &&
+      !label &&
```

Comment:
This would be a more thorough check for label text. Do we change the logic for checking a label to something more like that in case the developer passes a component without text/number as `label`?

https://github.com/FB-PLP/traject/blob/2a29bc21ae73061891bcc044761f98c20659bd03/app/assets/javascripts/v2/core/EDSCandidates/button.tsx#L200-L213

Reply:
that's a good instinct! IIRC a lot of that logic was added specifically to handle icons awhile back. Now that Andrew's added a11y checking at the Icon level (i.e. informative icons must have text titles), I think a lighter level of checking here is fine